source/jmap/types.d

```diff
@@ -279,94 +268,52 @@ version(SIL):
+        auto invocation = Invocation.set(type, activeAccountId(), invocationId, ifInState, create, update, destroy_, additionalArguments);
         auto request = JmapRequest(listCapabilities(), [invocation], null);
         return post(request);
     }
 
-    Variable set(string type, string ifInState = null, SilStruct create = null, SilStruct update = null, string[] destroy_ = null, SilStruct additionalArguments = null) {
-        return setRaw(type, ifInState, create, update, destroy_, additionalArguments).deserialize!Variable;
-    }
-
     Variable setEmail(string ifInState = null, SilStruct create = null, SilStruct update = null, string[] destroy_ = null, SilStruct additionalArguments = null) {
         return set("Email", ifInState, create, update, destroy_, additionalArguments);
     }
 
-
-    Asdf copyRaw(string type, string fromAccountId, string ifFromInState = null, string ifInState = null, SilStruct create = null, bool onSuccessDestroyOriginal = false, string destroyFromIfInState = null, SilStruct additionalArguments = null) {
-        import std.algorithm : map;
-        import std.array : array;
+    JsonAlgebraic copy(string type, string fromAccountId, string ifFromInState = null, string ifInState = null, StringMap!JsonAlgebraic create = null, bool onSuccessDestroyOriginal = false, string destroyFromIfInState = null, StringMap!JsonAlgebraic additionalArguments = null) {
         auto invocationId = "12345678";
-        auto createAsdf = parseJson(serializeJson(Variable(create)));
-        auto invocation = Invocation.copy(type, fromAccountId, invocationId, ifFromInState, activeAccountId, ifInState, createAsdf, onSuccessDestroyOriginal, destroyFromIfInState);
+        auto invocation = Invocation.copy(type, fromAccountId, invocationId, ifFromInState, activeAccountId, ifInState, create, onSuccessDestroyOriginal, destroyFromIfInState);
         auto request = JmapRequest(listCapabilities(), [invocation], null);
         return post(request);
     }
 
-    Variable copy(string type, string fromAccountId, string ifFromInState = null, string ifInState = null, SilStruct create = null, bool onSuccessDestroyOriginal = false, string destroyFromIfInState = null, SilStruct additionalArguments = null) {
-        return copyRaw(type, fromAccountId, ifFromInState, ifInState, create, onSuccessDestroyOriginal, destroyFromIfInState, additionalArguments).deserialize!Variable;
-    }
-
-
-    Asdf queryRaw(string type, Variable filter, Variable sort, int position, string anchor = null, int anchorOffset = 0, Nullable!uint limit = (Nullable!uint).init, bool calculateTotal = false, SilStruct additionalArguments = null) {
-        import std.algorithm : map;
-        import std.array : array;
+    JsonAlgebraic query(string type, JsonAlgebraic filter, StringMap!JsonAlgebraic sort, int position, string anchor = null, int anchorOffset = 0, Nullable!uint limit = null, bool calculateTotal = false, StringMap!JsonAlgebraic additionalArguments = null) {
```

Comment:
I've seen that you've changed all `sort` params from `Variable` to `SilStruct` (`StringMap!JsonAlgebraic`). I take it that was fully intentional?

Reply:
According to https://jmap.io/spec-core.html#query, `sort` is an array of objects.

Reply:
thank you!